cpp/src/parquet/arrow/writer.cc

```diff
@@ -403,17 +410,31 @@ class FileWriterImpl : public FileWriter {
     }
 
     auto WriteBatch = [&](int64_t offset, int64_t size) {
+      std::vector<std::unique_ptr<ArrowColumnWriterV2>> writers;
+
       int column_index_start = 0;
       for (int i = 0; i < batch.num_columns(); i++) {
-        ChunkedArray chunkedArray(batch.column(i));
+        ChunkedArray chunkedArray{batch.column(i)};
         ARROW_ASSIGN_OR_RAISE(
             std::unique_ptr<ArrowColumnWriterV2> writer,
             ArrowColumnWriterV2::Make(chunkedArray, offset, size, schema_manifest_,
                                       row_group_writer_, column_index_start));
-        RETURN_NOT_OK(writer->Write(&column_write_context_));
         column_index_start += writer->leaf_count();
+        writers.emplace_back(std::move(writer));
```

Comment:
I understand it's to unify use-thread and non-thread code, but it's looks not ideal to always create such a vector for non-thread case.

Reply:
Agreed. Let me refactor it to share common logic and avoid unnecessary vector creation.